File: numberGuessGame.java
import java.lang.Math;
import java.util.Scanner;

public class NumberGuessGame {
    int randomInt;
    int attempts;
    int lowerBound;
    int upperBound;
    String gameState;
    boolean gameOver;

    public NumberGuessGame(){
        resetGame();
    }

    void guess(String strGuess){
        try{
            int guess = Integer.parseInt(strGuess);
            if (guess <= lowerBound || guess >= upperBound) {
                this.gameState = String.format("Your guess is out of bounds. Enter a value between %d and %d%n", lowerBound, upperBound);
                return;
            }
            this.attempts++;
            if (guess < randomInt) { /* Too low */
                this.lowerBound = guess;
                this.gameState = String.format("Too low! Enter a value between %d and %d%n", lowerBound, upperBound);
            } else if (guess > randomInt) { /* Too high */
                this.upperBound = guess;
                this.gameState = String.format("Too high! Enter a value between %d and %d%n", lowerBound, upperBound);
            } else { /* Right value */
                if (this.attempts == 1) {
                    this.gameState = ("Amazing! You got it right in one attempt!");
                } else {
                    this.gameState = String.format("You got it right! Great work! You did it in %d attempts%n", attempts);
                }
                gameOver = true;
            }
            return;
        }
        catch(NumberFormatException e){
            this.gameState = String.format("Invalid input! Enter a value between %d and %d%n", lowerBound, upperBound);
        }
    }

    public String getState(){
        return gameState;
    }

    public String getAttempts(){
        return "You have made " + Integer.toString(attempts) + " guesses.";
    }

    public void resetGame(){
        randomInt = (int) (Math.floor(Math.random() * 100) + 1);
        attempts = 0;
        lowerBound = 1;
        upperBound = 100;
        gameState = "Enter a value between 1 and 100";
        gameOver = false;
    }
}

File: NumberGuessGame.py
from random import random

class NumberGuessGame:
    def __init__(self):
        self.randomInt = int(random()*100)
        self.attempts = 0
        self.lowerBound = 1
        self.upperBound = 100
        self.gameState = "Enter a value between 1 and 100"
        self.gameOver = False

    def guess(self,val):
        # generate some random numbers 
            try:
                guess = int(val)
                if guess < self.lowerBound or guess > self.upperBound:
                    self.gameState = "That guess is outside the bounds. Guess a number between " + str(self.lowerBound) + " and " + str(self.upperBound)
                    return
                self.attempts += 1
                if guess < self.randomInt:   #Guess too low
                    self.lowerBound = guess
                    self.gameState = "Too low! Guess a number between " + str(self.lowerBound) + " and " + str(self.upperBound)
                elif guess > self.randomInt: #Guess too low
                    self.upperBound = guess
                    self.gameState = "Too high! Guess a number between " + str(self.lowerBound) + " and " + str(self.upperBound)
                else:                   #Guess is correct
                    self.gameState = "That's the right number! Great work! It took you " + str(self.attempts) + " attempt(s) to guess it."
                    self.gameOver = True
            except ValueError:
                self.gameState = "Invalid input. Please guess a number between " + str(self.lowerBound) + " and " + str(self.upperBound)

    def getState(self):
        return self.gameState

    def getAttempts(self):
        return "You have made " + str(self.attempts) + " guesses."

    def resetGame(self):
        self.randomInt = int(random()*100)
        self.attempts = 0
        self.lowerBound = 1
        self.upperBound = 100
        self.gameState = "Enter a value between 1 and 100"
        self.gameOver = False
